### PR TITLE
Replace requirements for extras require

### PR DESCRIPTION
**Pull request by Rodrigo-Tenorio**
Following @RobertRosca 's suggestion in https://github.com/openjournals/joss-reviews/issues/3000#issuecomment-792999393
> Pretty subjective, but I think having all the extra requirements under extras_require would be nice, then you could have some for docs, style, test, etc... which makes development and contribution a bit easier, also makes the CI scripts cleaner since everything can be a pip install ".[style,test,...]" command instead of using separate requirements files

ditch `requirements-dev.txt` and `requirements.txt` in favor of `setup.py`'s `install_requires`and `extras_require`.

- [x] Include `extras_require` in `setup.py` and remove `requirements-dev.txt`.
- [x] Implement the usage of `extras_require` into the CI action.
- [x] Remove unnecessary usage of `requirements.txt` in Dockerfile.
- [x] Remove `requirements.txt` (after all, why should we keep it?).
- [x] Update bin/setup-dev-tools.sh
- [x] Document the usage of `extras_require` in the README.

Edit:

- [x] Include `readthedocs` dependencies into `extras_require` as well.

### CODE REVIEW

**Comment by dbkeitel** [comment]
Makes sense as far as the extras go. (Do you have a docs link handy for how this is supposed to be done canonically?)

But I think `requirements.txt` is required for conda-forge...? Though I'll have to double-check the docs for that, too.

**Comment by Rodrigo-Tenorio** [comment]
Here are the docs for extra dependencies: https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html

Essentially, if the user wants the extra dependencies labeled as `dev`: `pip install pyfstat[dev]`.

**Comment by Rodrigo-Tenorio** [comment]
@dbkeitel I see no `requirements.txt` in  https://github.com/PyFstat/staged-recipes/blob/PyFstat/recipes/PyFstat/meta.yaml ; in fact, it looks like dependencies are hard-coded there (I seem to remember conda not updating dependencies by itself?).

**Comment by dbkeitel** [comment]
> I see no requirements.txt in https://github.com/PyFstat/staged-recipes/blob/PyFstat/recipes/PyFstat/meta.yaml ; in fact, it looks like dependencies are hard-coded there (I seem to remember conda not updating dependencies by itself?).

You're probably right - I remembered they needed to be tracked separately, but probably not the right place. Let's try getting rid of it and seeing if the release workflow breaks anywhere.

File: .github/workflows/codecov.yml [hunk 1]
```diff
@@ -19,21 +19,10 @@ jobs:
         python-version: 3.8
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements.txt
```
Comment (dbkeitel) [inline .github/workflows/codecov.yml:23]: I love that you got rid of the multiple copies of the ephemerides setup! Rest of the workflow cleanup also looks good. But why removing the `python -m pip install --upgrade pip` here and in the others, that seems a reasonable precaution given how often pip deprecates and randomly breaks?

File: README.md [hunk 2]
```diff
@@ -170,25 +170,34 @@ pip install -r /PATH/TO/THIS/DIRECTORY/requirements.txt
...
+PyFstat manages optional dependencies through setuptool's `extras_require`.
+
+Available sets of optional dependencies are:
+
+* [chainconsumer](https://github.com/Samreay/ChainConsumer): Required to run some optional 
+plotting methods and some of the [example scripts](./examples).
+* [pycuda](https://pypi.org/project/pycuda/): Required for the `tCWFstatMapVersion=pycuda`
+  option of the `TransientGridSearch` class. (Note: Installing `pycuda` requires a working 
+  `nvcc` compiler in your path.)
+* style: Includes the [flake8](https://flake8.pycqa.org/en/latest/) linter
```
Comment (dbkeitel) [inline README.md:184]: use ``style`` and similar highlighting for all the tags to make clear they're the actual named tags and not just part of the README?

**Review by dbkeitel — CHANGES_REQUESTED** [review]
Thanks @Rodrigo-Tenorio that looks like a great clean-up. We'll just have to try if it all works out in the release workflow. I just left two small comments.

File: .github/workflows/codecov.yml (hunk 1 above)
Comment (Rodrigo-Tenorio) [reply .github/workflows/codecov.yml:23]: You are right, let's keep it.

File: README.md (hunk 2 above)
Comment (Rodrigo-Tenorio) [reply README.md]: I though links couldn't be highlighted. I was [`wrong`](https://www.youtube.com/watch?v=ROGOCFpcj2Q) :wink:

**Comment by Rodrigo-Tenorio** [comment]
@dbkeitel I've retracted the removal of `requirements.txt`since that's the way docker containers are currently built.

I propose to merge this and post-pone the removal of `requirements.txt`to whenever #88 is done.

**Comment by dbkeitel** [comment]
Does requirements.txt need to go back into the manifest then too, or does docker not care about that?

**Comment by Rodrigo-Tenorio** [comment]
> Does requirements.txt need to go back into the manifest then too, or does docker not care about that?

I was wrong: Docker installs the package by copying the repo and using pip (`pip install .`). 
I got confused by the `pip install -r requirements.txt` line, which is completely unnecessary if `setup.py` is properly set.
 
That line (which was kind of ubiquitous) must have bin a vestige from an ancient time.

**Comment by Rodrigo-Tenorio** [comment]
@dbkeitel  Ready to merge

**Review by dbkeitel — APPROVED** [review]

**Pull request merged**